cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -45,8 +45,13 @@ type InitConfiguration struct {
 	// NodeRegistration holds fields that relate to registering the new master node to the cluster
 	NodeRegistration NodeRegistrationOptions `json:"nodeRegistration,omitempty"`
 
-	// APIEndpoint represents the endpoint of the instance of the API server to be deployed on this node.
-	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
+	// LocalAPIEndpoint represents the endpoint of the API server instance that's deployed on this control plane node
+	// In HA setups, this differs from ClusterConfiguration.ControlPlaneEndpoint in the sense that ControlPlaneEndpoint
+	// is the global endpoint for the cluster, which then loadbalances the requests to each individual API server. This
+	// configuration object lets you customize what IP/DNS name and port the local API server advertises it's accessible
+	// on. By default, kubeadm tries to auto-detect the IP of the default interface and use that, but in case that process
+	// fails you may set the desired value here.
+	LocalAPIEndpoint APIEndpoint `json:"localAPIEndpoint,omitempty"`
```

Comment:
Was the old field shipped and persisted previously? If so, what will happen if a previously serialized config file is read and the old apiEndpoint field is discarded and the new localAPIEndpoint field is empty?

Reply:
yep, it was shipped as part of the previous config version v1alpha3.

if an old config YAML is read and the old field is found, it _should be_ converted to the new field.
https://github.com/kubernetes/kubernetes/pull/70761/files#diff-628349c40af3952df65127e8449c04e2R31

kubeadm 1.13 would support both v1alpha3 and v1beta1.

Reply:
I specifically meant if v1beta1 was shipped in a previous release with this field. If another version of the config was shipped, but handles conversion, that's fine

Reply:
yep, v1beta1 would be shipped for the first time in 1.13.